tests/unit/preprocessor/_multimodel/test_multimodel.py

```diff
@@ -1,267 +1,314 @@
+
+    result = multi_model_statistics(cubes, span, statistics)
+
+    assert isinstance(result, dict)
+    assert set(result.keys()) == set(statistics)
+
+    for i, statistic in enumerate(statistics):
+        result_cube = result[statistic]
+        expected_data = np.ma.array(expected[i], mask=False)
+        assert_array_almost_equal(result_cube.data, expected_data)
+
+
+@pytest.mark.parametrize('calendar1, calendar2, expected', (
+    ('360_day', '360_day', '360_day'),
+    ('365_day', '365_day', '365_day'),
+    ('365_day', '360_day', 'gregorian'),
+    ('360_day', '365_day', 'gregorian'),
+    ('gregorian', '365_day', 'gregorian'),
+    ('proleptic_gregorian', 'julian', 'gregorian'),
+    ('julian', '365_day', 'gregorian'),
+))
+def test_get_consistent_time_unit(calendar1, calendar2, expected):
+    """Test same calendar returned or default if calendars differ.
+
+    Expected behaviour: If the calendars are the same, return that one.
+    If the calendars are not the same, return 'gregorian'.
+    """
+    cubes = (
+        generate_cube_from_dates('monthly', calendar=calendar1),
+        generate_cube_from_dates('monthly', calendar=calendar2),
+    )
+
+    result = mm._get_consistent_time_unit(cubes)
+    assert result.calendar == expected
+
+
+@pytest.mark.parametrize('span', SPAN_OPTIONS)
+def test_edge_case_different_time_offsets(span):
+    cubes = (
+        generate_cube_from_dates('monthly',
+                                 '360_day',
+                                 offset='days since 1888-01-01'),
+        generate_cube_from_dates('monthly',
+                                 '360_day',
+                                 offset='days since 1899-01-01'),
+    )
+
+    statistic = 'min'
+    statistics = (statistic, )
+
+    result = multi_model_statistics(cubes, span, statistics)
+
+    result_cube = result[statistic]
+
+    time_coord = result_cube.coord('time')
+
+    assert time_coord.units.calendar == 'gregorian'
+    assert time_coord.units.origin == 'days since 1850-01-01'
+
+    desired = np.array((14., 45., 73.))
+    np.testing.assert_array_equal(time_coord.points, desired)
+
+    # input cubes are updated in-place
+    for cube in cubes:
+        np.testing.assert_array_equal(cube.coord('time').points, desired)
+
+
+def generate_cubes_with_non_overlapping_timecoords():
+    """Generate sample data where time coords do not overlap."""
+    time_points = range(1, 4)
+    dates1 = [datetime(1850, i, 15, 0, 0, 0) for i in time_points]
+    dates2 = [datetime(1950, i, 15, 0, 0, 0) for i in time_points]
+
+    return (
+        generate_cube_from_dates(dates1),
+        generate_cube_from_dates(dates2),
+    )
+
+
+@pytest.mark.xfail(reason='Multimodel statistics returns the original cubes.')
+def test_edge_case_time_no_overlap_fail():
+    """Test case when time coords do not overlap using span='overlap'.
+
+    Expected behaviour: `multi_model_statistics` should fail if time
+    points are not overlapping.
```

Comment:
I'm not sure if the desired behaviour here is to fail. I would say so if I was designing it from scratch, but maybe this is actually considered a 'feature' by some users/recipes?? Any thoughts on this @valeriupredoi @LisaBock

Reply:
@schlunma what do you think?

Reply:
My first intuition would also be to let it fail when there is no overlap, but I guess returning an empty cube (or no cube) could also be a reasonable option. I've never used that "feature", and I don't really know if anyone uses it. I would be fine with changing this behavior if it is not used anywhere.

Reply:
Okay, let's leave it like this then 👍

Reply:
remember the purpose of the test in general is to verify the integrity of the functional behavior, not to assume things on its own, so if the function fails if there is no overlap, the test should check for that :+1:

Reply:
Good point @valeriupredoi , the point of this PR is to establish the definition so that we have a better description of the functionality that we want. Currently there is no test that explicitly checks for this scenario, so the behaviour is not defined. 

Therefore we want to clarify the desired behaviour with this PR. Since this return statement is inconsistent with the normal output (a list is returned rather than a dict with statistics), I marked this with XFAIL to indicate a known problem (which will be fixed with #968 😁). 

If this is behaviour that we need with this function, that is also OK for me, then I will adjust the test to ensure that a list of cubes is given back if there is no overlap and update #968 accordingly.

Reply:
sure, good call! I am perfectly fine with this as it is here, discussing further in the PR is cool, the only bit that I think needs addressing is that array equality func below, other than that - good work here :beer: